src/routes/targeted-messaging/targeted-messaging.controller.ts

```diff
@@ -0,0 +1,100 @@
+import { TargetedSafeNotFoundError } from '@/domain/targeted-messaging/errors/targeted-safe-not-found.error';
+import {
+  CreateSubmissionDto,
+  CreateSubmissionDtoSchema,
+} from '@/routes/targeted-messaging/entities/create-submission.dto.entity';
+import { Submission } from '@/routes/targeted-messaging/entities/submission.entity';
+import { TargetedMessagingService } from '@/routes/targeted-messaging/targeted-messaging.service';
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { NumericStringSchema } from '@/validation/entities/schemas/numeric-string.schema';
+import { ValidationPipe } from '@/validation/pipes/validation.pipe';
+import {
+  Body,
+  Controller,
+  DefaultValuePipe,
+  Get,
+  HttpStatus,
+  NotFoundException,
+  Param,
+  ParseIntPipe,
+  Post,
+  Res,
+} from '@nestjs/common';
+import { ApiCreatedResponse, ApiOkResponse, ApiTags } from '@nestjs/swagger';
+import { Response } from 'express';
+
+@ApiTags('targeted-messaging')
+@Controller({
+  path: 'targeted-messaging',
+  version: '1',
+})
+export class TargetedMessagingController {
+  constructor(private readonly service: TargetedMessagingService) {}
+
+  @ApiOkResponse({ type: Submission })
+  @Get(
+    'outreaches/:outreachId/chains/:chainId/safes/:safeAddress/signers/:signerAddress/submissions',
+  )
+  async getSubmission(
+    @Res() res: Response,
+    @Param('outreachId', new DefaultValuePipe(0), ParseIntPipe)
+    outreachId: number,
+    @Param('chainId', new ValidationPipe(NumericStringSchema)) chainId: string,
+    @Param('safeAddress', new ValidationPipe(AddressSchema))
+    safeAddress: `0x${string}`,
+    @Param('signerAddress', new ValidationPipe(AddressSchema))
+    signerAddress: `0x${string}`,
+  ): Promise<Response> {
+    try {
+      const submission = await this.service.getSubmission({
+        outreachId,
+        chainId,
+        safeAddress,
+        signerAddress,
+      });
+      return res
+        .status(HttpStatus.OK)
+        .header('Cache-Control', 'no-cache')
+        .json(submission);
+    } catch (err) {
+      if (err instanceof TargetedSafeNotFoundError) {
+        return res
+          .status(HttpStatus.NO_CONTENT)
+          .header('Cache-Control', 'no-cache')
+          .json({});
+      }
+      throw err;
+    }
+  }
+
+  @ApiCreatedResponse({ type: Submission })
+  @Post(
+    'outreaches/:outreachId/chains/:chainId/safes/:safeAddress/signers/:signerAddress/submissions',
+  )
+  async createSubmission(
+    @Param('outreachId', new DefaultValuePipe(0), ParseIntPipe)
```

Comment:
As above regarding the default value.

Reply:
Done in [3707402](https://github.com/safe-global/safe-client-gateway/pull/1989/commits/37074023fa76e41a4731987b0fe2b5ba31111eef)

Thanks!